File: daily/402.移掉k位数字.py
#
# @lc app=leetcode.cn id=402 lang=python3
#
# [402] 移掉K位数字
#

# @lc code=start
class Solution:
    def removeKdigits(self, num: str, k: int) -> str:
        for _ in range(k):
            minN = num;
            for i in range(10):
                newNum = num.replace(str(i), '', 1)
                minN = newNum if len(newNum) < len(minN) or newNum < minN else minN
            num = minN.lstrip('0')
        return num if num else '0'
        
# @lc code=end

File: daily/402.移掉k位数字.js
/*
 * @lc app=leetcode.cn id=402 lang=javascript
 *
 * [402] 移掉K位数字
 */

// @lc code=start
/**
 * @param {string} num
 * @param {number} k
 * @return {string}
 */
var removeKdigits = function(num, k) {
  while(k--) {
    let min = num
    for(let i = 0; i < 10; ++i) {
      const newNum = num.replace(i, '')
      min = newNum.length < min.length || newNum < min ? newNum : min
    }
    num = min.replace(/^0/g, '')
  }
  return num && num !== 'NaN' ? num : '0'
};
// @lc code=end
